### ISSUE TITLE
Source Control no longer works on Ubuntu 20.04

### ISSUE THREAD

**Issue by xinthose** [microsoft/vscode#147797]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.66.2
- OS Version: Ubuntu 20.04.4 LTS

Steps to Reproduce:

1. open a visual studio workspace
2. click the source control button
3. all that is shown is "Initialize Repository" and "Publish to GitHub".  

I have a .git folder in the same directory as the workspace.  Running `git init` on the directory, or doing a fresh checkout does not help.  This same folder/repository has worked for years with VS Code Source Control without issues until last week.  I tried reverting the version back to March, but that did not help either.

**Comment by lszomoru** [comment]
Have you recently upgraded your version of Git? What version of Git are you using?

**Comment by xinthose** [comment]
@lszomoru I am not sure.  I do update the packages when Ubuntu gives me the popup occasionally.  git version: 2.25.1

**Comment by xinthose** [comment]
@lszomoru OK.  Here is the output:
```
[2022-04-21T18:19:30.361Z] Validating found git in: git
[2022-04-21T18:19:30.567Z] Using git 2.25.1 from git
```
The directory does exist.  I am using `portalogic_station.workspace`:
```
root@linux-laptop:~/Documents/github/portalogic/portalogic_station# ls -la
total 44
drwxrwxrwx  6 root    root    4096 Apr 20 14:24 .
drwxrwxrwx 10 elemech elemech 4096 Apr 20 14:24 ..
drwxrwxrwx 13 root    root    4096 Apr 20 14:24 back_end
drwxrwxrwx  4 root    root    4096 Apr 20 14:24 client
drwxrwxrwx  8 root    root    4096 Apr 21 08:05 .git
-rwxrwxrwx  1 root    root     407 Apr 20 14:24 .gitattributes
-rwxrwxrwx  1 root    root    1327 Apr 20 14:24 .gitignore
-rwxrwxrwx  1 root    root    2000 Apr 20 14:24 portalogic_station.code-workspace
-rwxrwxrwx  1 root    root       5 Apr 20 14:24 .prettierignore
-rwxrwxrwx  1 root    root     372 Apr 20 14:24 README.md
drwxrwxrwx  2 root    root    4096 Apr 20 14:24 .vscode
root@linux-laptop:~/Documents/github/portalogic/portalogic_station# ls .git
branches  config  description  HEAD  hooks  index  info  logs  objects  packed-refs  refs
```

**Comment by just-trey** [comment]
I have the same issue here is my output
`[2022-04-22T00:13:06.031Z] Validating found git in: git
[2022-04-22T00:13:06.048Z] Using git 2.32.0 from git
[2022-04-22T00:14:36.216Z] > git init [2ms]`

**Comment by rysmith504** [comment]
i am having the same issue as well but I am running ubuntu on windows

**Comment by SalmanOz** [comment]
> i am having the same issue as well but I am running ubuntu on windows

same and, having the same issue. I click the initialize repository and choose the my project folder but nothing happens.

**Comment by xinthose** [comment]
Possibly related: https://github.com/microsoft/vscode/issues/147923.

**Comment by xinthose** [comment]
I suppose Microsoft does not care about fixing this seeing as how there are 5k+ open issues.  Reverting back to version Nov 2021 allows GitHub source control to work again: https://code.visualstudio.com/updates/v1_63.

**Comment by Andrew-HT** [comment]
I am having the same issue. 

VS Code Version: 1.66.2
OS Version: Ubuntu 18.04.6 LTS (in VirtualBox v6.1.22)

I reverted to VSCode version 1.65.2 and the issue was present there too.

> I suppose Microsoft does not care about fixing this seeing as how there are 5k+ open issues. Reverting back to version Nov 2021 allows GitHub source control to work again: https://code.visualstudio.com/updates/v1_63. You have to make sure the repository has been checked out with a personal access code token and run `chmod -R 777 .` on the directory first.

Unfortunately, this workaround didn't work for me. I switched to version 1.63.2 of VSCode, git cloned my repo and authenticated with my personal access token, re-set the remote set-url origin with the personal access token, chmod, git init in the directory, restarted VSCode - no dice. I still have the same source control screen where the "Initialize Repository" button does nothing. 

I also attempted to go through the "Publish to GitHub" method and that did nothing either.

![VirtualBoxVM_UWLXAURHZL](https://user-images.githubusercontent.com/47784635/166825225-33752d1e-cf72-4269-ae71-4cdd375501f4.gif)
(Ignore where the cursor is, not sure why the capture has my cursor offset from where it actually is.)

**Comment by lszomoru** [comment]
Apologies for not getting back to you on this issue until now. The latest Insiders release (2022-05-16), contains logging improvements, as well as additional logging to the code that detects git repositories which will hopefully track down the root cause of the problem. Could you please:
1. Enable verbose logging as documented on [this wiki](https://github.com/microsoft/vscode/wiki/Git-&-SCM-Issues#log-level-setting)
2. Open the folder/workspace that contains the git repository
3. Open the git output channel and share the contents of it

**Comment by Andrew-HT** [comment]
Installed version 1.68.0-insiders and enabled Trace level logging.
Here's what I get:

> [2022-05-19T18:18:39.256Z] > git init [7ms]
> [2022-05-19T18:18:39.819Z] [trace] Opening repository: myrepo
> [2022-05-19T18:18:39.833Z] > git rev-parse --show-toplevel [4ms]
> [2022-05-19T18:18:39.833Z] fatal: unsafe repository ('myrepo' is owned by someone else)
> To add an exception for this directory, call:
> 
> 	git config --global --add safe.directory myrepo
> 	
> [2022-05-19T18:18:39.834Z] [trace] Opening repository for path='myrepo' failed; ex=Failed to execute git {
>   "exitCode": 128,
>   "gitCommand": "rev-parse",
>   "stdout": "",
>   "stderr": "fatal: unsafe repository ('myrepo' is owned by someone else)\nTo add an exception for this directory, call:\n\n\tgit config --global --add safe.directory myrepo\n"
> }

I already tried running the add safe.directory command and I don't notice any change. Same error in the log appears. From Googling this, it looks like it's related to the recent Git vulnerability?

https://github.blog/2022-04-12-git-security-vulnerability-announced/

Hope this helps.

**Comment by xinthose** [comment]
I was able to get the latest version of VS Code (`1.67.2`) to work on Ubuntu 22.04 with git.  I had to run this command first as non root `git config --global --add safe.directory /path/to/my/repo`.

**Comment by lszomoru** [comment]
Duplicate of #157159. This will be included in VS Code 1.74.
